### ISSUE TITLE
"Networking_Error" description is confusing when applied to a server response

### ISSUE THREAD

**Issue by bboreham** [jcmturner/gokrb5#161]
I'm trying to debug something else, which is almost certainly my own mistake, but along the way got confused by this message:

```
2018/07/06 15:52:34 [Root cause: Networking_Error] Networking_Error: TGS Exchange Error: issue sending TGS_REQ to KDC: KRB Error: (7) KDC_ERR_S_PRINCIPAL_UNKNOWN Server not found in Kerberos database - LOOKING_UP_SERVER
```

Referring to [this code](https://github.com/jcmturner/gokrb5/blob/master/client/TGSExchange.go#L29-L32):
```
	r, err := cl.SendToKDC(b, kdcRealm)
	if err != nil {
		return tgsReq, tgsRep, krberror.Errorf(err, krberror.NetworkingError, "TGS Exchange Error: issue sending TGS_REQ to KDC")
	}
```

`sendKDCUDP()` and `sendKDCTDP()` return various kinds of errors up to the point where they get a response from the server, all of which could fairly be guessed to be "networking errrors".

Once they have read a response and unmarshalled it into a `KRBError` struct this description is confusing, to me at least.

In other words, an error return from `checkForKRBError()` should not be tagged as "Networking_Error"
